apis/secretversion/v1alpha1/secretversion_types.go

```diff
@@ -0,0 +1,146 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// SecretVersionState gives the state of a [SecretVersion][google.cloud.secretmanager.v1.SecretVersion], indicating if it can be accessed.
+type SecretVersionState int32
+
+const (
+	// SecretVersionSTATEUNSPECIFIED represents that SecretVersion state is not specified. This value is unused and invalid.
+	SecretVersionSTATEUNSPECIFIED SecretVersionState = 0
+	// SecretVersionENABLED represents that SecretVersion state  may be accessed.
+	SecretVersionENABLED SecretVersionState = 1
+	// SecretVersionDISABLED represents that SecretVersion state may not be accessed, but the secret data
+	// is still available and can be placed back into the [ENABLED] state.
+	SecretVersionDISABLED SecretVersionState = 2
+	// SecretVersionDESTROYED represents that SecretVersion state is destroyed and the secret data is no longer
+	// stored. A version may not leave this state once entered.
+	SecretVersionDESTROYED SecretVersionState = 3
+)
+
+// SecretVersionStateName is to map integers with states
+var SecretVersionStateName = map[int32]string{
+	0: "STATE_UNSPECIFIED",
+	1: "ENABLED",
+	2: "DISABLED",
+	3: "DESTROYED",
+}
+
+// SecretVersionStateValue is to map states with integers
+var SecretVersionStateValue = map[string]int32{
+	"STATE_UNSPECIFIED": 0,
+	"ENABLED":           1,
+	"DISABLED":          2,
+	"DESTROYED":         3,
+}
+
+// SecretVersionParameters defines parameters for a desired Secret Manager's secret version.
+type SecretVersionParameters struct {
+	// SecretRef refers to the secret object(GCP) created in Kubernetes
+	// Required
+	SecretRef string `json:"secretRef,omitempty"`
+
+	// Payload is the secret payload of the [SecretVersion][google.cloud.secretmanager.v1.SecretVersion].
+	// Either this or KubeSecretRef should be provided
+	// +optional
+	Payload *SecretPayload `json:"payload,omitempty"`
+
+	// DesiredSecretVersionState is the desired state the end user wants for the secret version
+	// SecretVersionENABLED represents that SecretVersion state  may be accessed.
+	// SecretVersionENABLED SecretVersionState = 1
+	// SecretVersionDISABLED represents that SecretVersion state may not be accessed, but the secret data
+	// is still available and can be placed back into the [ENABLED] state.
+	// SecretVersionDISABLED SecretVersionState = 2
+	// SecretVersionDESTROYED represents that SecretVersion state is destroyed and the secret data is no longer
+	// stored. A version may not leave this state once entered.
+	// SecretVersionDESTROYED SecretVersionState = 3
+	// +kubebuilder:validation:Enum=1;2;3
```

Comment:
I wonder if we should instead make this `ENABLED,DISABLED,...` as that would likely map to the experience folks would expect here.

Reply:
Done!